pkg/server/plugin/upstreamauthority/vault/vault_client_test.go

```diff
@@ -163,30 +163,44 @@ func (vcs *VaultClientSuite) Test_NewAuthenticatedClient_TokenAuth() {
 	vcs.fakeVaultServer.LookupSelfResponseCode = 200
 	for _, c := range []struct {
 		name      string
+		token     string
 		response  []byte
 		reusable  bool
 		namespace string
+		wantError bool
```

Comment:
I think that in general I prefer to have an `err string` here as an expected error that can be used to match with the error obtained.  Test cases that don't expect errors would have an empty string. Then you can do something like
```
			if c.err != "" {
				vcs.Require().Equal(err.Error(), c.err)
			} ...
```
in the assertions to check for the error. I think that allows us to catch some unexpected behavior if the error thrown is different than what we expect.

Reply:
Thanks for review!  
Update: 5e0fd1f